### PR TITLE
Authenticate to BAR using a Managed Identity

### PR DESCRIPTION
**Pull request by dkurepa**
https://github.com/dotnet/arcade-services/issues/3516

### Release Note Category
- [ ] Feature changes/additions 
- [ ] Bug fixes
- [x] Internal Infrastructure Improvements

### Release Note Description
Authenticate to BAR using Managed Identity

### CODE REVIEW

File: src/ProductConstructionService/ProductConstructionService.Api/Configuration/PcsConfiguration.cs [hunk 1]
```diff
@@ -53,6 +54,11 @@ public static string GetRequiredValue(this IConfiguration configuration, string
 
         string databaseConnectionString = builder.Configuration.GetRequiredValue(DatabaseConnectionString);
 
+        if (appendUserId)
```
Comment (premun) [inline src/ProductConstructionService/ProductConstructionService.Api/Configuration/PcsConfiguration.cs:57]: It's unclear where `appendUserId` would be appended from the scope of the whole method (when you're calling it and that it's connected to SQL).

How about instead, we do `.Replace()` for a placeholder in the connection string and either the placeholder is present or not based on the appsettings value?

**Comment by premun** [comment]
How do you know the changes work? The test deployments all seem to fail? Or they fail after they deploy?

**Comment by dkurepa** [comment]
> How do you know the changes work? The test deployments all seem to fail? Or they fail after they deploy?

The deployment itself is working, so is the service, the reason why it's read is because of Authentication issues, will look into that separately.
I tested it by modifying the status controller to query something from the DB and it worked fine

File: src/ProductConstructionService/ProductConstructionService.Api/Configuration/PcsConfiguration.cs (hunk 1 above)
Comment (dkurepa) [reply src/ProductConstructionService/ProductConstructionService.Api/Configuration/PcsConfiguration.cs]: I like that, it's elegant, done

File: src/Maestro/SubscriptionActorService/.config/settings.Production.json [hunk 2]
```diff
@@ -4,5 +4,8 @@
     "ProductConstructionService": {
         "Uri": "https://github.com/dotnet/arcade-services/issues/3183",
         "Token": "TODO https://github.com/dotnet/arcade-services/issues/3183"
+    },
+    "BuildAssetRegistry": {
+        "ConnectionString": "Data Source=tcp:maestro-prod-server.database.windows.net,1433; Initial Catalog=BuildAssetRegistry; Authentication=Active Directory Managed Identity; Persist Security Info=False; MultipleActiveResultSets=True; Connect Timeout=30; Encrypt=True; TrustServerCertificate=False;"
```
Comment (premun) [inline src/Maestro/SubscriptionActorService/.config/settings.Production.json:9]: Just double-checking - we don't need the ID in Maestro?

**Review by premun — APPROVED** [review]

**Pull request merged**